audformat/core/database.py

```diff
@@ -758,6 +759,11 @@ def join_field(
 
         # join tables
         for other in others:
+            for misc_id, misc in other.misc_tables.items():
```

Comment:
I think it makes sense to first update `MiscTable` objects as they may be used in schemes that are referenced by `Table` objects. Within the `MiscTable` objects we probably have to update `MiscTable` objects used in schemes first, since those schemes may be used by other `MiscTable` objects.... but I'm not 100% sure :)

Reply:
I guess, would be nice to add a comment here in code that we need to update misc tables first.

Reply:
> probably update MiscTable objects used in schemes first, since those schemes may be used by other MiscTable objects

I guess even this will not be sufficient as inside a misc table used as a scheme you can also define a scheme by a misc table. So I'm afraid we need something like a dependency resolution to figure out which misc table has to be updated first.

Reply:
I created https://github.com/audeering/audformat/issues/258 to cover the dependency stuff, which should be handled in a different pull request.

Reply:
I added a comment.